backend/app/api/v1/endpoints/auth.py

```diff
@@ -0,0 +1,343 @@
+
+from app.core.dependency import (
+    check_authorization,
+    check_existing_email,
+    check_existing_username,
+    check_user_active,
+    check_user_permission,
+)
+from app.core.security import (
+    create_access_token,
+    generate_reset_token,
+    get_current_user,
+    get_current_user_role,
+    hash_password,
+    send_reset_email,
+    verify_old_password,
+    verify_reset_token,
+    verify_token,
+)
+from app.core.service import send_verification_email
+from app.db.crud import (
+    create_in_db,
+    fetch_data_by_id,
+    update_instance,
+    update_instance_fields,
+)
+from app.db.database import get_db
+from app.model.base_model import User
+from app.schema.auth_schema import UserInResponse
+from app.util.hash import async_hash_password, verify_password
+from logger import log
+
+router = APIRouter(prefix="/auth", tags=["Authentication:"])
+
+templates = Jinja2Templates(directory="app/templates")
+
+
+def test_create_user(test_client):
+    print("Executing test_create_user")
+    data = {
+        "username": "test_user",
+        "email": "test@example.com",
+        "password": "password123",
+        "role": "user",
+        "first_name": "Test",
+        "last_name": "User",
+        "contact_number": "1234567890",
+        "gender": "male"
+    }
+    response = test_client.post("/api/v1/auth/create-user", data=data)
+    print(f"Response status code: {response.status_code}")
+    print(f"Response content: {response.content}")
+    assert response.status_code == 201
+    response_json = response.json()
+    assert "id" in response_json
+    assert response_json["email"] == "test@example.com"
+
+
+
+@router.post("/verify")
+async def verify_email(
+    request: Request,
+    email: str = Form(...),
+    v_token: str = Form(...),
+    db: AsyncSession = Depends(get_db),
+):
+    log.info(f"Attempting to verify email: {email}")
+    verification_result = verify_token(email, v_token)
+
+    if verification_result:
+        user = await db.execute(select(User).filter(User.email == email))
+        user_instance = user.scalar_one()
+        user_instance.is_active = True
+        await db.commit()
+        log.info(f"Email {email} verified successfully")
+    else:
+        log.warning(f"Failed to verify email: {email}")
+
+    return templates.TemplateResponse(
+        "verification_result.html",
+        {"request": request, "verification_result": verification_result},
+    )
+
+
+@router.get(
+    "/user/{id}",
+    status_code=status.HTTP_200_OK,
+    response_model=UserInResponse,
+)
+async def get_user(
+    id: int,
+    db: AsyncSession = Depends(get_db),
+    user_id: int = Depends(get_current_user),
+    admin: str = Depends(get_current_user_role),
+):
+    try:
+        user = await fetch_data_by_id(db, User, id)
+        permission = await check_user_permission(user_id, admin, id)
+        if permission:
+            log.success(f"User with id {id} fetched successfully")
+            return user
```

Comment:
Use the FastAPI provided response classes to formulate the response object. This will make the code readable and better to handle difference status codes. 

Check this for reference:
https://fastapi.tiangolo.com/tutorial/response-model/

Reply:
Have added pydantic class for plain text messages so that there can be a response model for each API endpoint, even if it returns just a string message.

class:
```
class Message(BaseModel):
    message: str
```

API-implementation:
```
@router.post(
    "/logout",
    status_code=status.HTTP_200_OK,
    response_model=Message,
    description="User logout",
)
```

```
@router.post(
    "/create-user",
    status_code=status.HTTP_201_CREATED,
    response_model=UserInResponse,
    description="Create a new user",
)
```